components/table/filterDropdown.tsx

```diff
@@ -108,11 +109,17 @@ export default class FilterMenu<T> extends React.Component<FilterMenuProps<T>, F
   }
 
   confirmFilter() {
-    if (this.state.selectedKeys !== this.props.selectedKeys) {
-      this.props.confirmFilter(this.props.column, this.state.selectedKeys);
+    const { selectedKeys } = this.state;
+
+    if (this.hasSelectedKeys() && !shallowequal(selectedKeys, this.props.selectedKeys)) {
```

Comment:
Don't need `this.hasSelectedKeys()`, if `state.selectedKeys` and `props.selectedKeys` are both empty, they are shallow equal, otherwise, if one is empty and the other one is not, they are absolutely not shallow equal.